sklearn/linear_model/_omp.py

```diff
@@ -769,7 +769,9 @@ def fit(self, X, y):
         self : object
             Returns an instance of self.
         """
-        X, y = validate_data(self, X, y, multi_output=True, y_numeric=True)
+        X, y = validate_data(
+            self, X, y, multi_output=True, y_numeric=True, dtype=FLOAT_DTYPES
+        )
         n_features = X.shape[1]
 
         X, y, X_offset, y_offset, X_scale, Gram, Xy = _pre_fit(
```

Comment:
I think we should set `check_gram=False` here because `self.precompute` can only be a bool or "auto", so the gram matrix is always computed in `_pre_fit` and we can skip the check.

By the way I noticed that the docstring for `precompute` says
```
Whether to use a precomputed Gram and Xy matrix to speed up
calculations. Improves performance when :term:`n_targets` or
:term:`n_samples` is very large. Note that if you already have such
matrices, you can pass them directly to the fit method.
```
The last sentence is wrong. `fit` only accepts X and y, no other args. Would you mind removing the last sentence in this PR ?